### ISSUE TITLE
Vertical Arabic

### ISSUE THREAD

**Issue by ntounsi** [w3c/alreq#102]
(Following issue https://github.com/w3c/i18n-drafts/issues/81)

Independently from body texts that run normally vertically (e.g. CJK, Mongolian), small Arabic text can be set vertically, e.g. on a book spine or in table header. 

Although the flow direction may depend on writing style, countries or what ever, it might be desirable to have text read from top to bottom

Example-1) Vertical Arabic text running from top to bottom, and Latin from bottom to top (book spine).
https://app.box.com/s/i4j6xx6kn6gu5nt2ha5ynno7qemoxlq4

`writing-mode: sideways-lr;` appear appropriate for this case. Implemented only in Firefox.
(BTW:  sideways value "[is at-risk and may be dropped](https://www.w3.org/TR/css-writing-modes-3/#valdef-writing-mode-sideways-rl)" from CSS3!

Example-2) Vertical Arabic, letters upright (isolated form)

http://photo.elcinema.com.s3.amazonaws.com/uploads/_310x310_dec41d25dd424bae652d87c7c50339116861040b15b863c7e70c8ed9b48c9e12.jpg

http://farm4.static.flickr.com/3277/2830543747_de05b47f02.jpg

http://1.bp.blogspot.com/-46crVsYMaSw/VLICB1smWZI/AAAAAAAAzGQ/AbwLJoh_IQE/s1600/560671_433062960083202_2034873025_n.jpg

`text-orientation : upright; ` is doing this.

In Firefox, Arabic run from top to bottom (in parallel to Latin, by the way!..)
In Safari and Chrome Arabic run from bottom to top.

**Comment by behnam** [comment]
From discussion on weekly meeting:
We want two separate sections for the following cases:
1) sideways text and reading direction,
2) vertical upright text, including CJK matters.

**Comment by shervinafshar** [comment]
ACTION-89 is open for this and Najib is working on a small draft regarding the topic.

**Comment by behnam** [comment]
https://github.com/w3c/alreq/pull/122

**Comment by behnam** [comment]
Comments from the meeting:
* Let's reorder the three methods to have book-spine first, and the CJK case second?
* Instead of bullets, using sections (with a short title) for the methods.
* How about we rename `vertical-en.png` and `vertical-fr.png` to something more descriptive of the text effect, like `vertical-facing-left.png` and `vertical-facing-right.png`, or `vertical-rotated-left.png` and `vertical-rotated-right.png`
* Have Arabic text to be the "first line" in both boxes, to make it easier for audience to see the box rotation logic in action.
* "ISSUE 1" is just a note, not an issue; also it's describing behavior of a specific technology. "ISSUE 2" and "ISSUE 3" are fine.
* Regarding the third method, we better keep it shorter for now, until we have move/better evidence of common practices.

We landed the PR. @r12a wants to do some editing on the text. After that, @ntounsi can address the comments in a new PR.

**Comment by r12a** [comment]
After my PR of yesterday, the following changes still need to be addressed (by Najib):
* How about we rename `vertical-en.png` and `vertical-fr.png` to something more descriptive of the text effect, like `vertical-facing-left.png` and `vertical-facing-right.png`, or `vertical-rotated-left.png` and `vertical-rotated-right.png`
* Have Arabic text to be the "first line" in both boxes, to make it easier for audience to see the box rotation logic in action.

**Comment by behnam** [comment]
Continuing the discussion on behavior in vertical lines...

From http://www.unicode.org/reports/tr9/#Vertical_Text

> In the case of vertical line orientation, the Bidirectional Algorithm is still used to determine the levels of the text. However, these levels are not used to reorder the text, because the characters are usually ordered uniformly from top to bottom. Instead, the levels are used to determine the rotation of the text. Sometimes vertical lines follow a vertical baseline in which each character is oriented as normal (with no rotation), with characters ordered from top to bottom whether they are Hebrew, numbers, or Latin. When setting text using the Arabic script in vertical lines, it is more common to employ a horizontal baseline that is rotated by 90° counterclockwise so that the characters are ordered from top to bottom. Latin text and numbers may be rotated 90° clockwise so that the characters are also ordered from top to bottom.
> 
> The Bidirectional Algorithm is used when some characters are ordered from bottom to top. For example, this happens with a mixture of Arabic and Latin glyphs when all the glyphs are rotated uniformly 90° clockwise. The Unicode Standard does not specify whether text is presented horizontally or vertically, or whether text is rotated. That is left up to higher-level protocols.

Therefore, Unicode recommends here that when there's only one text (strong chars only) direction, the Arabic should be written top-to-bottom, so glyph rotations should be 90º CCW. This contradicts the behavior recommended by latest CSS specs, and what we have in the document at the moment.

What do you think?

**Comment by r12a** [comment]
I wonder what real life use cases they base this on. Perhaps we should ask them.

**Comment by r12a** [comment]
Btw, there seems to be no record, but i believe Najib completed his two remaining actions already. See http://w3c.github.io/alreq/#h_vertical_text

**Comment by ntounsi** [comment]
There was no action for that, but a PR https://github.com/w3c/alreq/pull/126

**Comment by r12a** [comment]
See https://github.com/w3c/alreq/issues/140 for a continuation of the Unicode question.

Otherwise, Najib's work is complete.
